config/config.go

```diff
@@ -639,47 +644,85 @@ func getStakingTLSCert(v *viper.Viper) (tls.Certificate, error) {
 	}
 }
 
-func getStakingSigner(v *viper.Viper) (bls.Signer, error) {
-	if v.GetBool(StakingEphemeralSignerEnabledKey) {
-		key, err := localsigner.New()
+func getStakingSigner(ctx context.Context, v *viper.Viper) (bls.Signer, error) {
+	ephemeralSignerEnabled := v.GetBool(StakingEphemeralSignerEnabledKey)
+	contentKeyExists := v.IsSet(StakingSignerKeyContentKey)
+	keyPathIsSet := v.IsSet(StakingSignerKeyPathKey)
+	rpcSignerURLExists := v.IsSet(StakingRPCSignerKey)
+
+	signingKeyPath := getExpandedArg(v, StakingSignerKeyPathKey)
+	_, err := os.Stat(signingKeyPath)
+	keyFileNotFound := errors.Is(err, fs.ErrNotExist)
+
+	switch {
+
+	case ephemeralSignerEnabled && !contentKeyExists && !keyPathIsSet && !rpcSignerURLExists:
+		signer, err := localsigner.New()
 		if err != nil {
-			return nil, fmt.Errorf("couldn't generate ephemeral signing key: %w", err)
+			return nil, fmt.Errorf("couldn't generate ephemeral signing signer: %w", err)
 		}
-		return key, nil
-	}
 
-	if v.IsSet(StakingSignerKeyContentKey) {
+		return signer, nil
+
+	case !ephemeralSignerEnabled && contentKeyExists && !keyPathIsSet && !rpcSignerURLExists:
 		signerKeyRawContent := v.GetString(StakingSignerKeyContentKey)
 		signerKeyContent, err := base64.StdEncoding.DecodeString(signerKeyRawContent)
+
```

Comment:
nit: unneeded diff, generally errors are checked on the following line without an extra line break